### ISSUE TITLE
Use case without sysimage

### ISSUE THREAD

**Issue by fonsp** [MichaelHatherly/BundledWebResources.jl#53]
Hey Michael!

This look really cool, and thanks for your work on RelocatablePackages.jl! Me and @disberd wonder if this package could also help with our use case:

We have a package that is used by Pluto users that uses some JS libs from the web to work. We can import this library directly with JS as a URL, in the html that the package displays. Like here: https://github.com/JuliaPluto/PlutoUI.jl/blob/1474af51881211d472b04dc4151895812c0cdee1/src/RangeSlider.jl#L130

The problem is that the package does not work without internet. 
- One workaround is to download the file manually and put it in the package source. E.g. [Bonito.jl](https://github.com/SimonDanisch/Bonito.jl/tree/master/js_dependencies)
- A cleaner option is from @disberd: a [separate repository](https://github.com/disberd/PlotlyArtifactsESM/releases) that downloads, bundles and uploads into a github release artifact. The release artifact can be [used in Artifacts.toml](https://github.com/JuliaPluto/PlutoPlotly.jl/blob/main/Artifacts.toml). 
- In Pluto, we have a GHA that bundles `main` and force pushes the `dist` folder into a new branch: [`release`](https://github.com/fonsp/Pluto.jl/tree/release), and we trigger tagbot on this branch instead of `main`.

What do you think about this problem, any ideas? I understand that BundledWebResources.jl solves this problem when your package is used via a sysimage or pkgimage, but do you think BundledWebResources.jl could also help with packages installed from registry?

Thanks!!
-fons

**Comment by MichaelHatherly** [comment]
Yeah, it might be an option for your use case, this wasn't made for just sysimages and relocatablilty. Mostly made it so that all resources get requested from the origin, not third-parties. The external resources get downloaded and cached in a scratchspace during precompilation, which ideally is happening just after `] add`ing so the assumption is that there is internet access at that point (though of course, there's situations where that won't even be true, in which case an artifact is probably the way to go.)

**Comment by fonsp** [comment]
Ah interesting! The scratch space system sounds great! But indeed the increased reliability of including assets in the initial install is something we really want! 

I think BundledWebResources.jl is a nice declarative API for resources and I can imagine that it will be easy to build tooling on top of it. Maybe strategies 2&3 from my post can be generalized and automated. 😄 A GHA could find all resources and include them in source (2) or as artifact (3), and BundledWebResources would then use this as its source, instead of the web+scratch.
